src/solver/implicitgemm_util.hpp

```diff
@@ -768,6 +768,37 @@ int amd_lds_write_max_length()
 
 constexpr std::size_t get_lds_max_number_of_byte() { return 65536; }
 
+static inline auto get_ck_common_compiler_flag(const ConvolutionContext& ctx)
+{
+    auto compiler_flag = std::string(" --std=c++14");
+
+    // HIP version
+    {
+        const auto hip_version = HipCompilerVersion();
+
+        auto hip_version_major = std::to_string(hip_version.major);
+        auto hip_version_minor = std::to_string(hip_version.minor);
+        auto hip_version_patch = std::to_string(hip_version.patch);
+
+        hip_version_minor = std::string(2 - hip_version_minor.length(), '0') + hip_version_minor;
+        hip_version_patch = std::string(5 - hip_version_patch.length(), '0') + hip_version_patch;
+
+        compiler_flag += std::string(" -DCK_HIP_VERSION=") + hip_version_major + hip_version_minor +
```

Comment:
#include <miopen/stringutils.hpp> for stringize macro.

Reply:
I got negative value for HIP_PACKAGE_VERSION_FLAT, it seems xxx,xxx,xxxxxx will go out of range

Reply:
It seems like 32-bit math is still used somewhere on the way, most likely in the clang command line.

Patch number could be very high, so I recommend using the following convention:
```C++
// 2 decimal digits for major and minor, 6 digits for patch number.
// Max 32 bit signed is 21,47,483647, so major should be limited to
// max 20 to avoid overflow. Max flat version number is
// Max number is 20,99,999999.
#if HIP_PACKAGE_VERSION_MAJOR > 20 || HIP_PACKAGE_VERSION_MAJOR > 99 || \
    HIP_PACKAGE_VERSION_PATCH > 999999
#error "Too big HIP version number(s)"
#endif
#define HIP_PACKAGE_VERSION_FLAT                                                \
    ((HIP_PACKAGE_VERSION_MAJOR * 100 + HIP_PACKAGE_VERSION_MINOR) * 1000000 + \
     HIP_PACKAGE_VERSION_PATCH)
```
This should work correctly with 32-bit signed math.

Reply:
oops I am wrong

Reply:
Overflow happens here: `((3 * 1000 + 7) * 100000 + 20315)`. How to fix:
- Replace `1000` with `1000ULL` in the HIP_PACKAGE_VERSION_FLAT macro so overflow wouldn't happen.
- Keep using `std::to_string` to pass a string literal instead of expression to the HIP compiler.

Reply:
No need to change from 3-3-6 digits to 2-2-6.

Reply:
fixed. Thanks